src/Data/Fin/Permutation/Components.agda

```diff
@@ -28,17 +28,32 @@ open ≡-Reasoning
 
 -- 'tranpose i j' swaps the places of 'i' and 'j'.
 
+{-
 transpose : ∀ {n} → Fin n → Fin n → Fin n → Fin n
-transpose i j k with does (k ≟ i)
-... | true  = j
-... | false with does (k ≟ j)
-...   | true  = i
-...   | false = k
+transpose i j k with does (k ≟ i) | does (k ≟ j)
+... | true | _ = j
+... | false | true = i
+... | false | false = k
+-}
+
+transpose : ∀ {n} → Fin n → Fin n → Fin n → Fin n
+transpose i j k = if does (k ≟ i)
+  then j
+  else if does (k ≟ j)
+    then i
+    else k
 
 --------------------------------------------------------------------------------
 --  Properties
 --------------------------------------------------------------------------------
 
+transpose-comm : ∀ {n} (i j : Fin  n) → transpose i j ≗ transpose j i
```

Comment:
This is now both here and in `Data.Fin.Permutation`. I think here is a better place for it.